File: examples/01-hello/js/brain.js
// A blocking chat: the full reply comes back in-band, so the only event the
// trace records is the `chat` call itself. `log` returns undefined, so the
// run reports `completed` instead of exiting with the reply text.
const provider = omw.provider.get("openai");
const messages = [{ role: "user", content: "say hi" }];
const reply = provider.chat("gpt-test", messages, []);
omw.host.log("info", reply.content);

File: examples/01-hello/wasm/brain.rs
// A blocking chat: the full reply comes back in-band, so the only event the
// trace records is the `chat` call itself. Returning `Ok(())` reports
// `completed` instead of exiting with the reply text.
#![no_main]

use omw_wasm_rust::{host, prelude::*};

omw_wasm_rust::brain!(|| {
  let provider = Provider::get("openai")?;
  let messages = [ChatMessage::user("say hi")];
  let reply = provider.chat("gpt-test", &messages, &[])?;
  host::info(reply.text().unwrap_or_default());
  Ok(())
});
